include/onnxruntime/core/common/allocator_mimalloc.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <mimalloc.h>
+
+#pragma warning(disable: 4100)
+
+template <class T>
+struct allocator_mimalloc {
+  typedef T value_type;
```

Comment:
Shouldn't a mimalloc type be defined by files outside of ORT (e.g. be included from cmake/external/mimalloc)? If ORT includes types for external components in the core files it will most likely become a mess.

The setup seems off as it's inserting mimalloc specific things into core ORT code, instead of creating a more generic mechanism whereby the default allocator for std::vector can be overridden. Something similar to [https://github.com/microsoft/onnxruntime/blob/3d75c1b6be50b552e2734298d1355668a4ed3cd9/include/onnxruntime/core/framework/allocator.h#L301-L307](https://github.com/microsoft/onnxruntime/blob/3d75c1b6be50b552e2734298d1355668a4ed3cd9/include/onnxruntime/core/framework/allocator.h#L301-L307) is what I would have expected.

Reply:
This is a good point - I'll approach the maintainer of mimalloc about adding this allocator to their library.